scripts/xtensa-build-all.sh

```diff
@@ -30,30 +33,71 @@ usage: xtensa-build.sh [options] platform(s)
        -u Force UP ARCH
        -d Enable debug build
        -c Interactive menuconfig
+       -o override config
        -k Use private key
        -v Verbose Makefile log
        -j n Set number of make build jobs. Jobs=#cores when no flag. \
 Infinte when not specified.
+	-m MEU path
+	-s openssl path
        Supported platforms ${SUPPORTED_PLATFORMS[*]}
 EOF
 }
 
+if [ $# -eq 0 ]
+then
+	print_usage
+	exit 1
+fi
+
 # parse the args
-while getopts "rudj:ckva" OPTION; do
+while getopts "rudj:ck:vao:m:s:" OPTION; do
         case "$OPTION" in
 		r) BUILD_ROM=yes ;;
 		u) BUILD_FORCE_UP=yes ;;
 		d) BUILD_DEBUG=yes ;;
 		j) BUILD_JOBS=$OPTARG ;;
 		c) MAKE_MENUCONFIG=yes ;;
-		k) USE_PRIVATE_KEY=yes ;;
+		o) OVERRIDE_CONFIG=$OPTARG ;;
+		k) PRIVATE_KEY=$OPTARG ;;
```

Comment:
Good catch, the commit subject and message should probably be louder that this change breaks "backward compatibility".

I bet @ranj063 removes the environment option because it is called `RIMAGE_PRIVATE_KEY` whereas using MEU involves... both `MEU_PRIVATE_KEY` and `RIMAGE_PRIVATE_KEY`?! I could really use some more light on this...

It should be easy to fallback on the `RIMAGE_PRIVATE_KEY` env variable when no `-k` option is passed with a bit more code but do we really need to keep this feature ? @xiulipan , @lgirdwood, others: are you aware of anyone using this convenience script to build for any platform with rimage and a non-default key?

Reply:
@xiulipan I feel that changing to require the path for the private key is more intuitive than ENV. @marc-hb what do you think?

Reply:
`-k` is more explicit than `$RIMAGE_PRIVATE_KEY` which makes troubleshooting issues much easier. It's also more tedious, there are pros and cons on both sides. It's easy to support both, it "just" requires more code.

The most important questions for me right now are a bit different, they are: 1. backward compability - do some users really need `$RIMAGE_PRIVATE_KEY`? and 2.  the simultaneous use of RIMAGE_PRIVATE_KEY + MEU_PRIVATE_KEY. See other comments.

Reply:
i've removed this commit

Reply:
@ranj063 @marc-hb CI is build with different key with this method. OK, so I do not need to update the scripts for Jenkins CI now.